lxd/storage/backend_lxd.go

```diff
@@ -518,8 +518,27 @@ func (b *lxdBackend) ensureInstanceSymlink(instanceType instancetype.Type, proje
 		return errors.New("Instance must not be snapshot")
 	}
 
+	// Defend against path traversal attacks.
+	if !shared.IsFileName(instanceName) {
+		return fmt.Errorf("Invalid instance name %q", instanceName)
+	}
+
+	if !shared.IsFileName(projectName) {
+		return fmt.Errorf("Invalid project name %q", projectName)
+	}
+
 	symlinkPath := InstancePath(instanceType, projectName, instanceName, false)
 
+	// Defend against path traversal attacks.
+	baseDir := shared.VarPath()
+	if !shared.IsPathWithinBaseDir(baseDir, symlinkPath) {
+		return fmt.Errorf("Invalid symlink path %q", symlinkPath)
+	}
+
+	if !shared.IsPathWithinBaseDir(baseDir, mountPath) {
+		return fmt.Errorf("Invalid mount path %q", mountPath)
+	}
```

Comment:
I'm wondering if those would not be redundant once the instance and project names were validated?